core/src/main/java/fj/P1.java

```diff
@@ -167,11 +132,57 @@ public B _1() {
      * @return A single P1 for the given array.
      */
     public static <A> P1<Array<A>> sequence(final Array<P1<A>> as) {
-        return new P1<Array<A>>() {
-            public Array<A> _1() {
-                return as.map(P1.<A>__1());
-            }
-        };
+        return P.lazy(u -> as.map(P1.<A>__1()));
+    }
+
+    /**
+     * Traversable instance of P1 for List
+     *
+     * @param f The function that takes A and produces a List<B> (non-deterministic result)
+     * @return A List of P1<B>
+     */
+    public <B> List<P1<B>> traverseList(final F<A, List<B>>  f){
+        return f.f(_1()).map(b -> P.p(b));
+    }
+
+    /**
+     * Traversable instance of P1 for Either
+     *
+     * @param f The function might successfully return B or might fail with an error notification of type X
+     * @return An Either of  P1<B>
+     */
+    public <B, X> Either<X, P1<B>> traverseEither(final F<A, Either<X, B>>  f){
+        return f.f(_1()).right().map(b -> P.p(b));
+    }
+
+    /**
+     * Traversable instance of P1 for Option
+     *
+     * @param f The function might successfully might fail
+     * @return An Option of  P1<B>
+     */
+    public <B> List<P1<B>> traverseOption(final F<A, List<B>>  f){
+        return f.f(_1()).map(b -> P.p(b));
+    }
+
```

Comment:
I think you have copy and pasted and forgotten to change the signature to include option here.